pkg/stanza/fileconsumer/internal/trie/trie_test.go

```diff
@@ -198,3 +198,359 @@ func runTest(t *testing.T, trie *Trie, tc *testCase) {
 		}
 	}
 }
+
+func TestTrieOpSequences(t *testing.T) {
+	opTree{
+		continuations: map[string]opTree{
+			"Found:ABC": opTree{ // First poll finds only ABC.
+				ops: []testOp{
+					put("ABC"),
+				},
+				continuations: map[string]opTree{
+					"Done:ABC": opTree{ // Finish reading ABC and remove from trie
+						ops: []testOp{
+							del("ABC", true, "was just added"),
+							has("ABC", false, "was just deleted"),
+						},
+					},
+					"Found:ABCDEF": opTree{ // Next poll finds ABCDEF
+						ops: []testOp{
+							has("ABCDEF", true, "recognize ABC w/ DEF appended"),
+							put("ABCDEF"), // TODO HasKey returns true, so how do we know to call this?
+							has("ABC", false, "should push ABC down to ABCDEF"),
+						},
+						// TODO When we started reading this file, we identified it as ABC.
+						//      However, we have an updated understanding that it is ABCDEF.
+						//      Should we only have to delete ABC, or only ABCDEF, or both?
+						continuations: map[string]opTree{
+							"DeleteAs:ABC": opTree{ // Done reading the file, remove it as ABC
+								ops: []testOp{
+									del("ABC", false, "should have been pushed down when ABCDEF was added"), // TODO incorrectly returns true
+									has("ABCDEF", true, "should not have been deleted"),
+								},
+								continuations: map[string]opTree{
+									"DeleteAs:ABCDEF": opTree{ // Also remove it as ABCDEF
+										ops: []testOp{
+											del("ABCDEF", true, "just confirmed it exists"), // TODO this fails to delete
+										},
+									},
+								},
+							},
+							"DeleteAs:ABCDEF": opTree{ // Done reading the file, remove it as ABC
+								ops: []testOp{
+									del("ABCDEF", true, "trying to delete ABC should not affect ABCDEF"),
+									has("ABC", true, "should not have been deleted"),
+								},
```

Comment:
This will return false, as we previously pushed down the value at [#219](https://github.com/VihasMakwana/opentelemetry-collector-contrib/pull/2/files#diff-bdd2c539982d6fc9548d9e953dab9e86076d8a7bec0b4287d0f7d06d31879b58R219)